daffodil-core/src/main/scala/org/apache/daffodil/core/dpath/Expression.scala

```diff
@@ -2295,6 +2295,18 @@ abstract class FunctionCallBase(
     res
   }
 
+  final def checkArgArray(): Unit = {
+    lazy val isArray = expressions.head match {
```

Comment:
This `checkArgArray` function is only used for `fn:count`, which only has one argument so it's fine to just check `expressions.head`, but without knowing that it could be confusing why this only checks head and not all of the function arguments. I wonder if this could be made more generic somehow? Maybe it takes an index into the expressions list and checks that that specific expression is an array? And then the usage in FnCountExpr would be `checkArgArray(0)`? Or maybe it takes an actual expression, so the usage would be `checkArgArray(args(0))`? I'm not sure if a similar check we do somewhere else that we could copy the pattern from?